coriolis/api/v1/migrations.py

```diff
@@ -106,11 +109,18 @@ def create(self, req, body):
                 raise exc.HTTPBadRequest(
                     explanation="Invalid destination "
                                 "environment: %s" % message)
+            is_valid, message = (
+                self._endpoints_api.validate_source_environment(
+                    context, origin_endpoint_id, source_environment))
+            if not is_valid:
```

Comment:
Could you please define a method/decorator in `coriolis.api.v1.utils` which abstracts away this `is_valid, message = ...` patter and apply it both here, in the Replicas file, and for the dest-env validation?